tests/scripts/local_monitoring_tests/rest_apis/test_update_metadata_profile.py

```diff
@@ -0,0 +1,420 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+import pytest
+import json
+import sys
+import copy
+import os
+import shutil
+
+sys.path.append("../../")
+
+from helpers.fixtures import *
+from helpers.kruize import *
+from helpers.utils import *
+from helpers.list_metadata_profiles_validate import *
+from helpers.list_metadata_profiles_schema import *
+from helpers.list_metadata_profiles_without_parameters_schema import *
+from helpers.import_metadata_json_validate import *
+from helpers.list_metadata_json_schema import *
+
+metadata_profile_dir = get_metadata_profile_dir()
+
+mandatory_fields = [
+    ("apiVersion", ERROR_500_STATUS_CODE, ERROR_STATUS),
+    ("kind", ERROR_500_STATUS_CODE, ERROR_STATUS),
+    ("metadata", ERROR_500_STATUS_CODE, ERROR_STATUS),
+    ("name", ERROR_500_STATUS_CODE, ERROR_STATUS),
+    ("datasource", ERROR_500_STATUS_CODE, ERROR_STATUS),
+    ("query_variables", ERROR_500_STATUS_CODE, ERROR_STATUS),
+    ("name", ERROR_500_STATUS_CODE, ERROR_STATUS),
+    ("value_type", ERROR_500_STATUS_CODE, ERROR_STATUS),
+    ("aggregation_functions", ERROR_500_STATUS_CODE, ERROR_STATUS),
+    ("function", ERROR_500_STATUS_CODE, ERROR_STATUS),
+    ("query", ERROR_500_STATUS_CODE, ERROR_STATUS)
+]
+
+
+@pytest.mark.sanity
+def test_update_metadata_profile(cluster_type):
+    """
+    Test Description: This test validates the response status code of updateMetadataProfile API by passing a
+    valid input for the json
+    """
+    input_json_file = metadata_profile_dir / 'cluster_metadata_local_monitoring.json'
+    form_kruize_url(cluster_type)
+
+    json_data = json.load(open(input_json_file))
+    metadata_profile_name = json_data['metadata']['name']
+
+    delete_and_create_metadata_profile()
+
+    response = list_metadata_profiles(name=metadata_profile_name)
+    metadata_profile_json = response.json()
+
+    assert response.status_code == SUCCESS_200_STATUS_CODE
+
+    # Validate the json against the json schema
+    errorMsg = validate_list_metadata_profiles_json(metadata_profile_json, list_metadata_profiles_schema)
+    assert errorMsg == ""
+
+    update_json_file = "../json_files/update_cluster_metadata_local_monitoring.json"
+
+    dsmetadata_json_file = "../json_files/import_metadata.json"
+    import_metadata_list_and_validate(dsmetadata_json_file, verbose="true")
+
+    # Update metadata profile using the specified json
+    response = update_metadata_profile(update_json_file, name=metadata_profile_name)
+    data = response.json()
+
+    assert response.status_code == SUCCESS_STATUS_CODE
+    assert data['status'] == SUCCESS_STATUS
+    assert data['message'] == UPDATE_METADATA_PROFILE_SUCCESS_MSG % metadata_profile_name
+
+    dsmetadata_json_file = "../json_files/import_metadata.json"
+
+    if cluster_type == "openshift":
+        import_metadata_list_and_validate(dsmetadata_json_file, verbose="true", validate_workload="true", 
```

Comment:
Validate the count of workloads & namespaces against what is specified in the profile. For example, just check if there is only a single workload & if the workload is kruize.